scholarly/_navigator.py

```diff
@@ -0,0 +1,195 @@
+                                   'https': _TOR_SOCK}
+            else:
+                session.proxies = _PROXIES
+
+            try:
+                _HEADERS['User-Agent'] = UserAgent().random
+
+                resp = session.get(pagerequest,
+                                   headers=_HEADERS,
+                                   cookies=_COOKIES,
+                                   timeout=_TIMEOUT)
+                
+                if resp.status_code == 200:
+                    if self._has_captcha(resp.text):
+                        raise Exception("Got a CAPTCHA. Retrying.")
+                    else:
+                        break
+                else:
+                    self.logger.info(f"""Got a response code {resp.status_code}. 
+                                    Retrying...""")
+                    raise
+
+            except Exception as e:
+                self.logger.info(f"Exception {e} while fetching page. Retrying.")
+                # Check if Tor is running and refresh it
+                self.logger.info("Refreshing Tor ID...")
+                self._refresh_tor_id()
+                session.close()
+        session.close()
+        return resp.text
+
+
+    def _tor_works(self) -> bool:
+        """ Checks if Tor is working"""
+        with requests.Session() as session:
+            session.proxies = {
+                'http': _TOR_SOCK,
+                'https': _TOR_SOCK
+            }
+            try:
+                #Changed to twitter so we dont ping google twice every time
+                resp = session.get("http://www.twitter.com")
+                self.logger.info("TOR Works!")
+                return resp.status_code == 200
+            except Exception as e:
+                self.logger.info("NO TOR :(")
+                return False
+
+
+    def _refresh_tor_id(self) -> bool:
+        try:
+            with Controller.from_port(port=_TOR_CONTROL) as controller:
+                controller.authenticate(password="")
+                controller.signal(Signal.NEWNYM)
+            return True
+        except Exception as e:
+            self.logger.info(f"Exception {e} while refreshing TOR. Retrying...")
+            return False
+
+
+    def use_proxy(self, http:str, https:str):
+        """ Routes scholarly through a proxy (e.g. tor).
+            Requires pysocks
+            Proxy must be running."""
+        self.logger.info("Enabling proxies: http=%r https=%r", http, https)
+        _PROXIES = {
+            "http": http,
+            "https": https,
+        }
+
+
+    def use_tor(self):
+        self.logger.info("Setting tor as the proxy")
+        self._use_proxy(http=_TOR_SOCK,
+                  https=_TOR_SOCK)
+
+
+    def _has_captcha(self, text:str) -> bool:
+        flags = ["Please show you're not a robot",
+                 "network may be sending automated queries",
+                 "have detected unusual traffic from your computer",
+                 "scholarly_captcha",
```

Comment:
Actually the image url for the CAPTCHA includes `/sorry/image` not `scholarly_captcha`